core/transport/src/grpc/job_master.proto

```diff
@@ -17,22 +17,30 @@ enum Status {
   COMPLETED = 5;
 }
 
-message TaskInfo {
-  optional int64 jobId = 1;
-  optional int32 taskId = 2;
-  optional Status status = 3;
-  optional string errorMessage = 4;
-  optional bytes result = 5;
+enum JobType {
+  PLAN = 1;
+  TASK = 2;
 }
 
+message JobUnused {}
+
+// next available id: 12
 message JobInfo {
   optional int64 id = 1;
   optional string errorMessage = 2;
-  repeated TaskInfo taskInfos = 3;
   optional Status status = 4;
-  optional string result = 5;
-  optional int64 lastStatusChangeMs = 6;
+  optional int64 lastUpdated = 6;
+  optional JobType type = 8;
+  optional bytes result = 9;
+
+  // Some jobs don't have these do not have these
   optional string name = 7;
+  optional int64 parentId = 10;
+  repeated JobInfo children = 11;
+
+  // Around for backwards compatibility
+  repeated JobUnused unused0 = 3; // formerly taskInfos
```

Comment:
Is it fine to replace the object type?

Reply:
Status: CREATED   Count: 0
Status: CANCELED  Count: 0
Status: FAILED    Count: 0
Status: RUNNING   Count: 0
Status: COMPLETED Count: 644

10 Most Recently Modified Jobs:
Timestamp: 11-04-2019 16:14:35:088       Id: 1572912790406       Name: Persist             Status: COMPLETED
Timestamp: 11-04-2019 16:14:35:088       Id: 1572912790407       Name: Persist             Status: COMPLETED
Timestamp: 11-04-2019 16:14:35:088       Id: 1572912790404       Name: Persist             Status: COMPLETED
Timestamp: 11-04-2019 16:14:35:088       Id: 1572912790405       Name: Persist             Status: COMPLETED
Timestamp: 11-04-2019 16:14:35:088       Id: 1572912790402       Name: Persist             Status: COMPLETED
Timestamp: 11-04-2019 16:14:35:088       Id: 1572912790403       Name: Persist             Status: COMPLETED
Timestamp: 11-04-2019 16:14:35:088       Id: 1572912790400       Name: Persist             Status: COMPLETED
Timestamp: 11-04-2019 16:14:35:088       Id: 1572912790401       Name: Persist             Status: COMPLETED
Timestamp: 11-04-2019 16:14:35:088       Id: 1572912790414       Name: Persist             Status: COMPLETED
Timestamp: 11-04-2019 16:14:35:088       Id: 1572912790415       Name: Persist             Status: COMPLETED

10 Most Recently Failed Jobs:

10 Longest Running Jobs:

➜  alluxiocp git:(master)

Yeah it appears to be fine.. (The server was launched from this branch.)